core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2640,4 +2640,13 @@ package object config {
       .version("4.0.0")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val DEFAULT_ML_INTERMEDIATE_STORAGE_LEVEL =
+    ConfigBuilder("spark.ml.defaultIntermediateStorageLevel")
```

Comment:
If this storageLevel really matters, I guess we should:
1, add `HasIntermediateStorageLevel` in `sharedParams` (in both scala and python);
2, make these implementations extend `HasIntermediateStorageLevel`;

Reply:
So you mean to make it configurable in programming API, just like we did in ALS, rather than configurations?

Reply:
yes, I personally prefer this way. To make other implementations consistent with ALS.